src/server/pegasus_mutation_duplicator.cpp

```diff
@@ -175,17 +175,18 @@ void pegasus_mutation_duplicator::duplicate(mutation_tuple_set muts, callback cb
         dsn::task_code rpc_code = std::get<1>(mut);
         dsn::blob raw_message = std::get<2>(mut);
         auto dreq = dsn::make_unique<dsn::apps::duplicate_request>();
-        uint64_t hash = get_hash_from_request(rpc_code, raw_message);
 
         if (rpc_code == dsn::apps::RPC_RRDB_RRDB_DUPLICATE) {
             // ignore if it is a DUPLICATE
+            continue;
```

Comment:
I just can't understand your updates in this function, could you please explain to me?

Reply:
Sure. I left a comment here. It was a bug. DUPLICATE can not participate in duplication like other normal writes.